amount.go

```diff
@@ -181,13 +217,28 @@ func (a Amount) RoundTo(digits uint8, mode RoundingMode) Amount {
 		RoundDown:     apd.RoundDown,
 	}
 	result := apd.New(0, 0)
-	ctx := apd.BaseContext.WithPrecision(16)
+	ctx := contextPrecision(a.number.NumDigits())
 	ctx.Rounding = extModes[mode]
 	ctx.Quantize(result, a.number, -int32(digits))
 
 	return Amount{result, a.currencyCode}
 }
 
+func contextPrecision(digits ...int64) *apd.Context {
```

Comment:
What is the reason behind introducing this logic? In which cases was the hardcoded 16 not enough?

Reply:
Testing with `math.MaxInt64` failed, returning `0` from a call to `Round()`